src/allomancy/java/leaf/cosmere/allomancy/common/manifestation/AllomancyIronSteel.java

```diff
@@ -541,4 +574,220 @@ public static void CheckRecipeForMetal(TagKey<Item> containsMetal, Recipe<?> rec
 			}
 		}
 	}
+
+	// set scan range of Lines Thread
+	public static void setScanRange(int range)
+	{
+		linesThread.setScanRange(range);
+	}
+
+	// this should be threaded to avoid lag spikes on the render thread when flaring metals
+	static class IronSteelLinesThread implements Runnable {
+		private static Thread t;
+		private static final Lock lock = new ReentrantLock();
+		private static ScanResult scanResult = new ScanResult();
+		private static int scanRange = 0;
+		private static boolean isStopping = false;
+
+		public ScanResult requestScanResult()
+		{
+			lock.lock();
+			try
+			{
+				return scanResult;
+			}
+			catch (Exception e)
+			{
+				e.printStackTrace();
+				lock.unlock();
+				return new ScanResult();	// empty ScanResult so it doesn't crash
+			}
+		}
+
+		public void releaseScanResult()
+		{
+			lock.unlock();
+		}
+
+		public void setScanRange(int range)
+		{
+			scanRange = range;
+		}
+
+		public void start()
+		{
+			if (t == null)
+			{
+				CosmereAPI.logger.info("Starting lines thread");
+				t = new Thread(this, "lines_thread");
+				t.start();
+			}
+		}
+
+		public void stop()
+		{
+			if (t != null && !isStopping)
+			{
+				isStopping = true;
+			}
+		}
+
+		private void setScanResult(ScanResult result)
+		{
+			scanResult = result;
+		}
+
+		@Override
+		@OnlyIn(Dist.CLIENT)
+		public void run() {
+			while (!isStopping) {
+				final Minecraft mc = Minecraft.getInstance();
+				final ProfilerFiller profiler = mc.getProfiler();
+				ScanResult nextScan;
+				LocalPlayer playerEntity = mc.player;
+				//only update box list every so often
+				if (playerEntity.tickCount % 15 != 0)
+				{
+					try
+					{
+						Thread.sleep(50);
+					}
+					catch (InterruptedException e)
+					{
+						throw new RuntimeException(e);
+					}
+					continue;
+				}
+				nextScan = new ScanResult();
+
+				//find all the things that we want to draw a line to from the player
+
+				//metal blocks
+				profiler.push("cosmere-getBlocksInRange");
+				{
+					BlockPos.withinManhattanStream(playerEntity.blockPosition(), scanRange, scanRange, scanRange)
+							.filter(blockPos ->
+							{
+								Block block = playerEntity.level.getBlockState(blockPos).getBlock();
+								final boolean validMetalBlock = block instanceof IHasMetalType iHasMetalType && iHasMetalType.getMetalType() != Metals.MetalType.ALUMINUM;
+								boolean isGood = validMetalBlock || containsMetal(block);
+
+								if (isGood)
+								{
+									try
+									{
+										Player player = Minecraft.getInstance().player;
+										Level level = Minecraft.getInstance().level;
+										BlockState endBlock = Objects.requireNonNull(level.getBlockState(blockPos));
```

Comment:
Somehow managed to crash my game when opening up the debug menu with graph via shift + f3.
![image](https://github.com/leafreynolds/cosmere/assets/80794784/dfa501b3-d0ab-49bb-9f4c-c66e84b41a47)

Running the game again didn't crash it the second time, but I suspect it might be a race condition, as I still got an exception in thread. This might be happening because we push/pop stuff on the profiler from a different thread. We may need to remove those from the thread :D 

![image](https://github.com/leafreynolds/cosmere/assets/80794784/87bb82ad-b232-4f92-856d-0d21e41a84e8)

Reply:
I'll put it in the start/stop methods instead then. Or should I remove it entirely? I'm not actually sure what it does, as I'm pretty new to Minecraft modding as a whole

Reply:
I know what it does now. I decided to merge the two profiler things into one, just called "cosmere-linesThread". I push it at the start of the while loop, and pop it at the end. After some professional debugging (spamming shift + f3 for a few seconds), it doesn't seem to crash anymore. I also can't see the thread on the pie chart, which is either really good, or it just isn't working.

Reply:
Looking good! I'll have a play around later today and see if I can get the profiler graph stuff to show. If it's just not usable in a thread, we can remove it.

Reply:
All right, can't find it in any of the menus. maybe it's incorrectly listing it as unspecified, but more likely it's just ignoring it since it's not from the same thread. It probably doesn't make sense to track it, since the profiler is specific to the thread it's tracking. 

Let's just get rid of the profiler stuff in the thread :D

Reply:
Alright! Gimme a sec and I'll have another commit pushed :)

Reply:
Also done!